packages/xstate-wallet/src/workflows/application.ts

```diff
@@ -135,7 +136,7 @@ const generateConfig = (
             ],
             JOIN_CHANNEL: {
               target: 'settingSite',
-              actions: [actions.sendJoinChannelResponse]
+              actions: [actions.assignRequestId, actions.sendJoinChannelResponse]
```

Comment:
The context here seems a bit off to me -- it suggests that the application workflow is responsible for a single request.

Reply:
I think the `application` workflow will only handle a `Create/JoinRequest` and a `ConcludeRequest` right now so perhaps we should store them separately on the `context` ?

Reply:
In any case, seems unrelated to this change.